### PR TITLE
Added subcommands to resource view.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This allows us to get textures from subcommands.

### CODE REVIEW

File: gapis/api/cmd_extras.go [hunk 1]
```diff
@@ -60,6 +61,20 @@ func (e *CmdExtras) Add(es ...CmdExtra) {
 	}
 }
 
+// CloneFromOrPanic clones on or more CmdExtras to the list of CmdExtras,
+// if there was an error, a panic is raised
+func (e *CmdExtras) CloneFromOrPanic(es ...CmdExtra) {
```
Comment (ben-clayton) [inline gapis/api/cmd_extras.go]: We've used `MustXXX()` for the no-error form of `XXX()` methods elsewhere

File: gapis/api/vulkan/vulkan_terminator.go [hunk 2]
```diff
@@ -312,7 +312,7 @@ func cutCommandBuffer(ctx context.Context, id api.CmdID,
 	// idx[3] is the secondary command buffer index inside a vkCmdExecuteCommands
 	// idx[4] is the secondary command inside the secondary command-buffer
 	submitCopy := cb.VkQueueSubmit(a.Queue, a.SubmitCount, a.PSubmits, a.Fence, a.Result)
-	submitCopy.Extras().Add(a.Extras().All()...)
+	submitCopy.Extras().CloneFromOrPanic(a.Extras().All()...)
```
Comment (ben-clayton) [inline gapis/api/vulkan/vulkan_terminator.go]: Why not just: `submitCopy := deep.MustClone(a).(*VkQueueSubmit)` ?

File: gapis/resolve/state.go [hunk 3]
```diff
@@ -58,11 +59,16 @@ func (r *GlobalStateResolvable) Resolve(ctx context.Context) (interface{}, error
 	if err != nil {
 		return nil, err
 	}
+
 	s, err := capture.NewState(ctx)
 	if err != nil {
 		return nil, err
 	}
 	for _, cmd := range cmds {
+		log.W(ctx, "State: %+v", cmd)
```
Comment (ben-clayton) [inline gapis/resolve/state.go]: No thank you!

Comment (AWoloszyn) [inline gapis/resolve/state.go]: Thanks! I missed that one.

File: gapis/api/vulkan/vulkan_terminator.go (hunk 2 above)
Comment (AWoloszyn) [inline gapis/api/vulkan/vulkan_terminator.go]: The only reason is that we use Add(a.Extras().All()....) A lot elsewhere, when we are not cloning an atom.

**Comment by Qining** [comment]
LGTM

File: gapis/api/vulkan/vulkan_terminator.go (hunk 2 above)
Comment (ben-clayton) [inline gapis/api/vulkan/vulkan_terminator.go]: Why not both?

File: gapis/api/cmd_extras.go [hunk 4]
```diff
@@ -60,6 +61,20 @@ func (e *CmdExtras) Add(es ...CmdExtra) {
 	}
 }
 
+// MustClone clones on or more CmdExtras to the list of CmdExtras,
```
Comment (ben-clayton) [inline gapis/api/cmd_extras.go]: one or more.

Comment (AWoloszyn) [inline gapis/api/cmd_extras.go]: Done

**Pull request merged**